app/Models/Course.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class Course extends Model
+{
+    use HasFactory, SoftDeletes;
+
+    protected $fillable = [
+        'course_name',
+        'slug_course',
+        'image',
+        'description',
+        'price',
+        'status',
+    ];
+
+    public function tags()
+    {
+        return $this->belongsToMany(Tag::class, 'course_tag', 'course_id', 'tag_id');
+    }
+
+    public function lessons()
+    {
+        return $this->hasMany(Lesson::class, 'course_id');
+    }
+
+    public function userCourse()
```

Comment:
teacherCourse này để lấy gì vậy?

Reply:
> userCourse này để lấy gì vậy?

Cái này để lấy những người tham gia khóa học ạ

Reply:
sao trong course lại lấy chính nó làm gì nhỉ? 
câu lệnh: `return $this->belongsToMany(User::class, 'user_course', 'course_id', 'user_id');` này trả về giá trị gì?